### DISCUSSION TITLE
Designing complex Filters for data

### DISCUSSION THREAD

**Discussion by Khushli** [twilio-labs/paste#1702] [General] [answered]
Hi team,

For enterprise insights, we have a need for designing multiple complex filters for data tables. I have created some solutions with a little spin on the existing components

1. I need a different type of combobox for the filter operators (e.g AND, OR, NOT)  to create a clear distinction between the operators and the filters. Here's what I have created. 
<img width="631" alt="Screen Shot 2021-07-27 at 6 09 32 PM" src="https://user-images.githubusercontent.com/87436655/127234044-5d6909b3-63ce-4753-b717-04c6f344da9e.png">

2. I wanted to ask about truncating options within a combobox. Which one option would you advise? 
<img width="848" alt="Screen Shot 2021-07-27 at 4 48 18 PM" src="https://user-images.githubusercontent.com/87436655/127234126-8bc244fa-15bb-4b8b-9574-e6d940d77a1b.png">

**Comment by TheSisb** [comment]
Hey @Khushli,

This kind of question would be best talked through together in an office hour. Would you be willing to sign up for one? You can find the link as a recent message in the #help-design-system slack channel.  

We've also started moodboarding some patterns - filter is one. It would be great if you could drop your thoughts and iterations in the filter moodboard on Figma: https://www.figma.com/file/6MlUZI1JHp8aIWE5JCGlm0/?node-id=8%3A16148

**Reply by Khushli** [reply]
Here is the link to the Figma file - https://www.figma.com/file/lT5YYQkHap6DS0W1jsYqZt/?node-id=667%3A20098

**Comment by SiTaggart** [comment, marked as answer]
Hi @Khushli 

To round out the Office hours, our preference would be something similar to the design below, using the disclosure.

![image](https://user-images.githubusercontent.com/368249/127696806-4dac8190-18d5-4c89-b4ab-dd617c93bd8a.png)

To recap the feedback:

- Please keep the labels above all input fields in the filtering panel. This is really important for accessibility.
- Please be mindful of choosing the correct UI control to fit the interaction, and not based on it's appearance. The correct control for the appropriate task is also really important for accessibility, you can then use other tools such as copy, spacing, alignment, borders and backgrounds to create hierarchy, visual relationships.

The OHs recording is [available here](https://www.loom.com/share/754fef78a91c4548ac355d5dc459c879) and the password is available in the [#help-design-systems](https://twilio.slack.com/archives/CBEQHHDLP/p1627672439003500) Slack channel
